Code/Tools/ProjectManager/Source/ProjectUtils.cpp

```diff
@@ -461,8 +462,22 @@ namespace O3DE::ProjectManager
             if (projectDirectory.exists())
             {
                 // Check if there is an actual project here or just force it
-                if (force || PythonBindingsInterface::Get()->GetProject(path).IsSuccess())
+                AZ::Outcome<ProjectInfo> pInfo = PythonBindingsInterface::Get()->GetProject(path);
+                if (force || pInfo.IsSuccess())
                 {
+                    //determine if we have a restricted directory to worry about
+                    if (!pInfo.GetValue().m_restricted.isEmpty())
+                    {
+                        QDir restrictedDirectory(QStandardPaths::standardLocations(QStandardPaths::HomeLocation).first());
+                        restrictedDirectory.cd("O3DE");
+                        restrictedDirectory.cd("Restricted");
+                        restrictedDirectory.cd("Projects");
+                        restrictedDirectory.cd(pInfo.GetValue().m_restricted);
```

Comment:
This approach is potentially dangerous is `pInfo.GetValue().m_restricted` has an error and actually does not exist. QDir's `cd` method will actually check the existence of a directory name and return true or false if it can traverse to that folder. Its also dangerous if someone manually deletes their `O3DE` folder in their $HOME folder. Since you are not checking the return type, it will trickle down to the `removeRecursively` function, and potentially delete all of the projects in the `$HOME/O3DE/Restricted/Projects` folder if the project folder name does not exist in there. Even worse, if they delete their `O3DE` folder manually, it might delete all the folders in the `$HOME` path entirely. You need to check the result of all the `cd` operations and only do the deletion if they all succeed